### PR TITLE
[all hosts] Add Glossary of terms article

### PR DESCRIPTION
**Pull request by alison-mk**
Reviewers: I plan to request a review from our editor **after** incorporating feedback from a first round of reviews by writers.

The internal [work item](https://office.visualstudio.com/OC/_workitems/edit/5326821) for this task includes background information on the concept and goals.

### CODE REVIEW

File: docs/resources/resources-glossary.md
Comment (AlexJerabek) [inline docs/resources/resources-glossary.md:116]: Maybe add a line like "It is used in the Microsoft 365 documentation to differentiate between modern Office Add-ins and older COM or VSTO add-ins."

File: docs/resources/resources-glossary.md
Comment (AlexJerabek) [inline docs/resources/resources-glossary.md:122]: Consider addressing the outdated nature of XLLs.

File: docs/resources/resources-glossary.md
Comment (AlexJerabek) [inline docs/resources/resources-glossary.md:128]: Since this refers to the command line argument, consider stylizing the casing accordingly.
```suggestion
The [Yeoman generator for Office Add-ins](https://github.com/OfficeDev/generator-office) uses the open source [Yeoman](https://github.com/yeoman/yo) tool to generate an Office Add-in via the command line. **yo office** is the command line argument that runs the Yeoman generator for Office Add-ins. The Office Add-ins quickstarts and tutorials use the Yeoman generator.
```

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md:102]: Thanks, good catch! I forgot the alphabet.

Comment (alison-mk) [reply docs/resources/resources-glossary.md:102]: Thanks, good catch! I forgot the alphabet.

**Review by alison-mk — COMMENTED** [review]
Thanks for the reviews! Really appreciate all your suggestions.

File: docs/toc.yml
Comment (ElizabethSamuel-MSFT) [inline docs/toc.yml:1341]: Since it's in the Office Add-ins documentation, it might suffice to simply say "Glossary"
```suggestion
  - name: Glossary
```

File: docs/resources/resources-glossary.md
Comment (ElizabethSamuel-MSFT) [reply docs/resources/resources-glossary.md:54]: I disagree about linking to this. It seems like an example M365 enterprise put together for their purpose.

File: docs/resources/resources-glossary.md
Comment (ElizabethSamuel-MSFT) [reply docs/resources/resources-glossary.md:58]: I thought so too but the [Specify Office hosts...](https://docs.microsoft.com/office/dev/add-ins/develop/specify-office-hosts-and-api-requirements) article says no (though it adds OneNote). [JSON SOT](https://github.com/OfficeDev/Office-Js-Requirement-Sets/blob/master/mapping/requirements_officejs.json) also doesn't note Word as supporting ContentApp.

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md:62]: My goal here is to reference "user-defined function" as a general programming concept, such as: https://en.wikipedia.org/wiki/User-defined_function. What if I change the sentence to "a custom function is a type of user-defined function in Excel"?

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md:90]: My understanding is that shared runtime ensures ribbon, custom function, and task pane code all run in the same lifecycle. 

@davidchesnut Could you provide feedback on the definition of **runtime** in this Glossary PR? 

Thanks!

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md:12]: I'll let our editor make the final call on this. I don't see any guidance in our style guide for how to format a glossary, but our style guide itself doesn't capitalize nouns in this context unless they're proper nouns. Term Studio also only capitalizes proper nouns. 

I think following the style of a typical dictionary makes the most sense and teaches our customers how to use the terms properly in context by distinguishing proper nouns from nouns.

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md:110]: I see your point, but since custom functions is never hyphenated I want to avoid it. Do you think italicizing "custom functions add-in" would help in this context?

File: docs/resources/resources-glossary.md
Comment (AlexJerabek) [reply docs/resources/resources-glossary.md:62]: What if we emphasize the add-ins model that custom functions have? How about "a **custom function** is a user-defined function that is packaged with an Excel add-in" (or something to that effect). If you don't want to go that route, I think I prefer your original text.

File: docs/resources/resources-glossary.md
Comment (Rick-Kirkham) [inline docs/resources/resources-glossary.md:58]: I find the term "surfaces" mysterious. These are embedded webviews aren't they?

File: docs/resources/resources-glossary.md
Comment (ElizabethSamuel-MSFT) [reply docs/resources/resources-glossary.md:58]: @alison-mk Also consider updating the Content Add-ins article even in a subsequent PR

File: docs/resources/resources-glossary.md
Comment (ElizabethSamuel-MSFT) [reply docs/resources/resources-glossary.md:110]: I suggested the hyphen based on how I understood what's said in the [MS style guide](https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=28765) re: noun modifiers

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md]: I will leave this as an explicit list, because my understanding is that Excel, Outlook, PowerPoint, and Word are the only applications that support task panes (and that Project, OneNote, and Visio do not support task panes).

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md]: I don't understand, can you elaborate? Does **client** differ from **application** and **host** in this way?

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md]: I don't feel strongly one way or the other about this, but I did pull the definition from the !IMPORTANT note at the top of this page: https://docs.microsoft.com/en-us/office/dev/add-ins/develop/office-javascript-api-object-model

File: docs/resources/resources-glossary.md
Comment (ElizabethSamuel-MSFT) [inline docs/resources/resources-glossary.md:20]: Maybe we can also include a link to the Outlook article...
```suggestion
**Add-in commands** are UI elements, such as buttons and menus, that extend the Office UI for your add-in. When users select an add-in command element, they initiate actions such as running JavaScript code or displaying the add-in in a task pane. Add-in commands let your add-in look and feel like a part of Office, which gives users more confidence in your add-in. See [Add-in commands for Excel, PowerPoint, and Word](../design/add-in-commands.md) and [Add-in commands for Outlook](../outlook/add-in-commands-for-outlook.md) to learn more.
```

File: docs/resources/resources-glossary.md
Comment (ElizabethSamuel-MSFT) [inline docs/resources/resources-glossary.md:78]:
```suggestion
In the Office Add-ins context, a **platform** usually refers to the operating system running an add-in. Platforms that support Office Add-ins include Windows, Mac, iPad, and web browsers.
```

File: docs/resources/resources-glossary.md
Comment (ElizabethSamuel-MSFT) [inline docs/resources/resources-glossary.md:100]:
```suggestion
A **tutorial** is a teaching aid designed to help people learn to use a product or procedure. In the Office Add-ins context, a tutorial guides an add-in developer through the complete add-in development process for a particular application, such as Excel. This involves following 20 or more steps and is a greater time investment than a [quick start](#quick-start).

See also: [quick start](#quick-start).
```

File: docs/resources/resources-glossary.md
Comment (AlexJerabek) [reply docs/resources/resources-glossary.md]: There's a distinction between **client** and **server** in most web services documentation. There isn't a distinction in our docs between Office clients, applications, and hosts, but there is a distinction between the end-user's Office client and the web service hosting the add-in. What you have is true; I was just looking for a place to inject more context about the add-ins model.

Feel free to ignore this comment if you feel **client** is usually used as a synonym for Office application in our docs.

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md]: I'm not familiar with the term "webview", and I don't see internet search results that consistently define this term. In term studio, I see the definition of webview is "A view that displays web content inside your application." Is this a term broadly understood by developers? I'm hesitant to use it, but I could use "webview" in this definition and then also define webview in the glossary. 

For reference, I pulled this definition: from https://docs.microsoft.com/en-us/office/dev/add-ins/design/content-add-ins

Comment (Rick-Kirkham) [reply docs/resources/resources-glossary.md]: > I'm not familiar with the term "webview", and I don't see internet search results that consistently define this term. In term studio, I see the definition of webview is "A view that displays web content inside your application." Is this a term broadly understood by developers? I'm hesitant to use it, but I could use "webview" in this definition and then also define webview in the glossary.
> 
> I agree that "surfaces" is ambiguous.
> 
> For reference, I pulled this definition: from https://docs.microsoft.com/en-us/office/dev/add-ins/design/content-add-ins

We use "webview" elsewhere in our docs. Sometimes we say "web control" or "web view control" or "embedded browser". People are familiar with it. The task pane is a webview. We should change that article to use "webview" too.

File: docs/resources/resources-glossary.md
Comment (AlexJerabek) [reply docs/resources/resources-glossary.md:122]: I like it. Let's tag @davidchesnut to double check.

File: docs/resources/resources-glossary.md
Comment (AlexJerabek) [inline docs/resources/resources-glossary.md:26]: I don't think Visio technically supports add-ins. The Visio JS APIs allow for Visio diagrams to be embedded in SharePoint.

File: docs/resources/resources-glossary.md
Comment (AlexJerabek) [inline docs/resources/resources-glossary.md:32]: Counter to the earlier point, Visio does have an application-specific API. In fact, that's all it has because it can't use the Common APIs (the exception being OfficeExtension APIs, which are technically part of the application-specific model, but that level of detail doesn't need to be here).

File: docs/resources/resources-glossary.md
Comment (AlexJerabek) [inline docs/resources/resources-glossary.md]: The callout to PowerPoint might be misleading here. WXP are in the same boat: they have application-specific APIs and also functionality from the Common APIs. Maybe it's worth mentioning that WXP have Common API functionality, but most of it has been replaced by the application-specific model?

**Review by AlexJerabek — APPROVED** [review]
Approving with a couple questions around Visio clarifications and communicating Common vs app-specific.

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md:32]: I'll add Visio to this list.

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md:26]: My inclination is to simply remove Visio from this list. Do you think that's reasonable?

File: docs/resources/resources-glossary.md
Comment (ElizabethSamuel-MSFT) [inline docs/resources/resources-glossary.md:54]:
```suggestion
**Content add-ins** are webviews, or web browser views, that are embedded directly into Excel, OneNote, or PowerPoint documents. Content add-ins give users access to interface controls that run code to modify documents or display data from a data source. Use content add-ins when you want to embed functionality directly into the document. See [Content Office Add-ins](../design/content-add-ins.md) to learn more.
```

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

File: docs/resources/resources-glossary.md
Comment (alison-mk) [reply docs/resources/resources-glossary.md:12]: @lindalu-MSFT Can you take a look at this conversation and provide feedback from an editorial perspective?

File: docs/resources/resources-glossary.md
Comment (lindalu-MSFT) [reply docs/resources/resources-glossary.md:121]: Other terms to consider: SaaS, deploy/centralized deployment, integrated apps, SAML applications, manifest, FRE

File: docs/resources/resources-glossary.md
Comment (lindalu-MSFT) [reply docs/resources/resources-glossary.md:12]: Initially, I thought I agreed with Alex that the terms should be capitalized in the headings. But then I located several glossaries that follow this guidance, so now I agree more with Alison.: "The glossary uses standard (dictionary-style) capitalization in the terms list: each glossary term should be lowercase unless it is treated as a proper noun, in which case it should be capitalized."

**Pull request merged**
